jenkins-plugin/pom.xml

```diff
@@ -391,48 +350,27 @@
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>git</artifactId>
-      <version>${git-plugin.version}</version>
       <classifier>tests</classifier>
       <scope>test</scope>
     </dependency>
   </dependencies>
   <dependencyManagement>
     <dependencies>
       <dependency>
-        <groupId>org.jenkins-ci.plugins</groupId>
-        <artifactId>mailer</artifactId>
-        <version>1.21</version>
-      </dependency>
-      <dependency>
-        <groupId>org.jenkins-ci.plugins</groupId>
-        <artifactId>scm-api</artifactId>
-        <version>${scm-api-plugin.version}</version>
-      </dependency>
-      <dependency>
-        <groupId>org.jenkins-ci.plugins</groupId>
-        <artifactId>ssh-credentials</artifactId>
-        <version>1.14</version>
-      </dependency>
-      <dependency>
-        <groupId>org.jenkins-ci.plugins</groupId>
-        <artifactId>apache-httpcomponents-client-4-api</artifactId>
-        <version>4.5.5-3.0</version>
-      </dependency>
-      <dependency>
-        <groupId>org.apache.ant</groupId>
-        <artifactId>ant</artifactId>
-        <version>1.9.2</version>
+        <groupId>io.jenkins.tools.bom</groupId>
+        <artifactId>bom-2.204.x</artifactId>
+        <version>11</version>
+        <scope>import</scope>
+        <type>pom</type>
       </dependency>
+      <!-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ -->
+      <!-- Fix RequireUpperBoundDeps dependencies -->
```

Comment:
Risky. Run the actual `*.jpi` as a sanity check, not just `JenkinsRule` and `hpi:run` which have a different class loading environment.

Reply:
ok I will do it @jglick 
I added the comment because I suppose it was the goal of all these depMgt entries (I removed everything and kept only what was required to build). Not sure if @cyrille-leclerc can confirm.
I will run some tests locally and also on our internal masters.
As I am not at all confident as new owner I will also probably publish an experimental release (not sure if a lot of people are using the incremental ones - AFAICS and understand this plugin is supposed to publish the incremental binaries)

Reply:
An experimental release is probably overkill, this sort of update is routine, just need to do an end-to-end test in a realistic class loading environment once to make sure the spy still works, Maven classes can link, etc.

Reply:
It's more about others commits done in master since the previous release. I didn't yet review all of them. Not sure if I will be confident enough in the changes. Our CI tests are instable and I am not sure if these are only "infrastructure" related issues.